src/components/policy/src/policy/include/policy/policy_manager_impl.h

```diff
@@ -187,6 +188,25 @@ class PolicyManagerImpl : public PolicyManager {
 
     virtual std::string RetrieveCertificate() const OVERRIDE;
 
+    class RetrySequenceTimewatch {
+     public:
+      RetrySequenceTimewatch(PolicyManagerImpl* ptr);
+      ~RetrySequenceTimewatch();
+      void Stop();
+      bool IsRunning();
+      void Start(uint32_t timeout);
+
+     private:
+      typedef timer::TimerThread<RetrySequenceTimewatch> Timer;
+      Timer timer_retry_sequence_;
+
+      PolicyManagerImpl* policy_manager_;
+
+     private:
+      RetrySequenceTimewatch();
```

Comment:
@AGritsevich I completely understand that constructor's parameter is **very** important for all universe. But why do you need the possibility to create object **without** this wonderful parameter from private section?
BTW constructor with one parameter should be `explicit`